### DISCUSSION TITLE
Teslamate and VWsFriend side by side

### DISCUSSION THREAD

**Discussion by sodgaard** [tillsteinbach/VWsFriend#208] [Q&A] [answered]
Hi,

I already have Teslamate up and running.
To have VWsFriend alongside I modified the script so the second instance of Grafana is named GrafanaVW and the port for VWsFriend is 3010 and GrafanaVW is 4010 (on the left side) and unchanged on the right side.

I can connect to VWsFriend and it seem to have connection to VWconnect.
However when trying to access GrafanaVW I get a blank screen and a timeout..

Any idea what this could be?
May I suggest that the standard install allows for them to be installed side-by-side :-)

**Comment by tillsteinbach** [comment]
Can you paste the file you edited?

**Reply by sodgaard** [reply]
`version: '3.8'
services:
  vwsfriend:
    image: tillsteinbach/vwsfriend:${VWSFRIEND_VERSION-latest}
    ports:
     - 4010:4000
    networks:
      backend:
        aliases:
          - vwsfriendbackend
    volumes:
      - vwsfriend_data:/config
    environment:
      - VWSFRIEND_USERNAME=${VWSFRIEND_USERNAME-admin}
      - VWSFRIEND_PASSWORD=${VWSFRIEND_PASSWORD-secret}
      - WECONNECT_USER
      - WECONNECT_PASSWORD
      - WECONNECT_INTERVAL
      - DATABASE_URL=postgresql://${DB_USER-admin}:${DB_PASSWORD-secret}@${DB_HOSTNAME-postgresdbbackend}:${DB_PORT-5432}/${DB_NAME-vwsfriend}
      - ADDITIONAL_PARAMETERS=--with-database --with-abrp -vv
    depends_on:
      postgresdb:
        condition: service_healthy
    restart: unless-stopped
    healthcheck:
      test: (wget -qO- http://localhost:4000/healthcheck | grep "ok" -q) || exit 1
      interval: 60s
      timeout: 10s
      retries: 3

  postgresdb:
    image: postgres:13
    networks:
      backend:
        aliases:
          - ${DB_HOSTNAME-postgresdbbackend}
    volumes:
      - postgresdb_data:/var/lib/postgresql/data
    user: postgres
    environment:
      - POSTGRES_USER=${DB_USER-admin}
      - POSTGRES_PASSWORD=${DB_PASSWORD-secret}
      - POSTGRES_DB=${DB_NAME-vwsfriend}
      - TZ=UTC
      - PGTZ=UTC
    command: -p ${DB_PORT-5432}
    restart: unless-stopped
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -p ${DB_PORT-5432} --username=${DB_USER-admin} --dbname=${DB_NAME-vwsfriend}"]
      interval: 10s
      timeout: 5s
      retries: 5

  grafanaVW:
    image: tillsteinbach/vwsfriend-grafana:${VWSFRIEND_GRAFANA_VERSION-latest}
    ports:
      - 3010:3000
    networks:
      bridge:
      backend:
        aliases:
          - grafanabackend
    volumes:
      - vwsfriend-grafana_data:/var/lib/grafana
    environment:
      - GF_SECURITY_ADMIN_USER=${VWSFRIEND_USERNAME-admin}
      - GF_SECURITY_ADMIN_PASSWORD=${VWSFRIEND_PASSWORD-secret}
      - GF_SERVER_HTTP_PORT=${GF_SERVER_HTTP_PORT-3000}
      - DB_USER=${DB_USER-admin}
      - DB_PASSWORD=${DB_PASSWORD-secret}
      - DB_HOSTNAME=${DB_HOSTNAME-postgresdbbackend}
      - DB_PORT=${DB_PORT-5432}
      - DB_NAME=${DB_NAME-vwsfriend}
      - VWSFRIEND_USERNAME=${VWSFRIEND_USERNAME-admin}
      - VWSFRIEND_PASSWORD=${VWSFRIEND_PASSWORD-secret}
      - VWSFRIEND_HOSTNAME=${VWSFRIEND_HOSTNAME-vwsfriendbackend}
      - VWSFRIEND_PORT=${VWSFRIEND_PORT-4000}
    restart: unless-stopped
    healthcheck:
      test: ["CMD-SHELL", "(wget -qO- http://localhost:${GF_SERVER_HTTP_PORT-3000}/api/health | grep '\"database\": \"ok\"' -q) || exit 1"]
      interval: 60s
      timeout: 10s
      retries: 3

networks:
  bridge:
  backend:

volumes:
  vwsfriend_data:
  postgresdb_data:
  vwsfriend-grafana_data:`

**Reply by sodgaard** [reply]
A screendump of the GrafanaVW:

![image](https://user-images.githubusercontent.com/44952678/161399591-501df8e3-e578-4e64-be00-004d65a3b187.png)

**Reply by sodgaard** [reply]
And of VWsFriend:
![image](https://user-images.githubusercontent.com/44952678/161399624-400d3e74-c857-494d-972d-f6bbc8fd4d67.png)

**Reply by sodgaard** [reply]
GrafanaVW is on networks docker_backend and docker_bridge
VWsFriend is on network docker_bridge

**Reply by sodgaard** [reply]
And here is the up and running VWsFriend:
![image](https://user-images.githubusercontent.com/44952678/161399973-adbc4f4e-2df5-46ed-8b18-f98fe6eaa032.png)

**Comment by sodgaard** [comment]
Here is the log of the Grafana container:
`docker_grafanaVW_1;

date,stream,content;

"2022-04-01 21:54:17,stdout,logger=http.server t=2022-04-01T21:54:17.21+0000 lvl=info msg=""HTTP Server Listen"" address=[::]:3000 protocol=http subUrl= socket=";

"2022-04-01 21:54:16,stdout,logger=ngalert.multiorg.alertmanager t=2022-04-01T21:54:16.39+0000 lvl=info msg=""starting MultiOrg Alertmanager""";

"2022-04-01 21:54:16,stdout,logger=ngalert t=2022-04-01T21:54:16.31+0000 lvl=info msg=""warming cache for startup""";
;
"2022-04-01 21:54:16,stdout,logger=provisioning.datasources t=2022-04-01T21:54:16.21+0000 lvl=info msg=""inserting datasource from configuration "" name=VWsFriend uid=P2EF84….........";
;
"2022-04-01 21:54:16,stdout,logger=provisioning.datasources t=2022-04-01T21:54:16.21+0000 lvl=info msg=""deleted datasource based on configuration"" name=VWsFriend";
;
"2022-04-01 21:54:15,stdout,logger=provisioning.datasources t=2022-04-01T21:54:15.92+0000 lvl=info msg=""inserting datasource from configuration "" name=""VWsFriend Live"" uid=P36D08….....";
;
"2022-04-01 21:54:15,stdout,logger=provisioning.datasources t=2022-04-01T21:54:15.92+0000 lvl=info msg=""deleted datasource based on configuration"" name=""VWsFriend Live""";
;
"2022-04-01 21:54:13,stdout,logger=live.push_http t=2022-04-01T21:54:13.75+0000 lvl=info msg=""Live Push Gateway initialization""";
;
"2022-04-01 21:54:13,stdout,logger=query_data t=2022-04-01T21:54:13.67+0000 lvl=info msg=""Query Service initialization""";
;
"2022-04-01 21:54:13,stdout,logger=plugin.manager t=2022-04-01T21:54:13.59+0000 lvl=info msg=""Plugin registered"" pluginId=natel-plotly-panel";
;
"2022-04-01 21:54:13,stdout,logger=plugin.manager t=2022-04-01T21:54:13.59+0000 lvl=info msg=""Plugin registered"" pluginId=marcusolsson-json-datasource";
;
"2022-04-01 21:54:13,stdout,logger=plugin.manager t=2022-04-01T21:54:13.59+0000 lvl=info msg=""Plugin registered"" pluginId=gapit-htmlgraphics-panel";
;
"2022-04-01 21:54:13,stdout,logger=plugin.manager t=2022-04-01T21:54:13.59+0000 lvl=info msg=""Plugin registered"" pluginId=pr0ps-trackmap-panel";
;
"2022-04-01 21:54:13,stdout,logger=plugin.manager t=2022-04-01T21:54:13.59+0000 lvl=info msg=""Plugin registered"" pluginId=panodata-map-panel";
;
"2022-04-01 21:54:13,stdout,logger=plugin.signature.validator t=2022-04-01T21:54:13.59+0000 lvl=warn msg=""Permitting unsigned plugin. This is not recommended"" pluginID=panodata-map-panel pluginDir=/var/lib/grafana/plugins/panodata-map-panel";
;
"2022-04-01 21:54:13,stdout,logger=plugin.manager t=2022-04-01T21:54:13.02+0000 lvl=info msg=""Plugin registered"" pluginId=input";
;
"2022-04-01 21:54:10,stdout,logger=migrator t=2022-04-01T21:54:10.49+0000 lvl=info msg=""migrations completed"" performed=0 skipped=381 duration=2.780377ms";
;
"2022-04-01 21:54:10,stdout,logger=migrator t=2022-04-01T21:54:10.46+0000 lvl=info msg=""Starting DB migrations""";
;
"2022-04-01 21:54:09,stdout,logger=sqlstore t=2022-04-01T21:54:09.96+0000 lvl=info msg=""Connecting to DB"" dbtype=sqlite3";
;
"2022-04-01 21:54:09,stdout,logger=settings t=2022-04-01T21:54:09.88+0000 lvl=info msg=""App mode production""";
;
"2022-04-01 21:54:09,stdout,logger=settings t=2022-04-01T21:54:09.88+0000 lvl=info msg=""Path Provisioning"" path=/etc/grafana/provisioning";
;
"2022-04-01 21:54:09,stdout,logger=settings t=2022-04-01T21:54:09.88+0000 lvl=info msg=""Path Plugins"" path=/var/lib/grafana/plugins";
;
"2022-04-01 21:54:09,stdout,logger=settings t=2022-04-01T21:54:09.88+0000 lvl=info msg=""Path Logs"" path=/var/log/grafana";
;
"2022-04-01 21:54:09,stdout,logger=settings t=2022-04-01T21:54:09.88+0000 lvl=info msg=""Path Data"" path=/var/lib/grafana";
;
"2022-04-01 21:54:09,stdout,logger=settings t=2022-04-01T21:54:09.88+0000 lvl=info msg=""Path Home"" path=/usr/share/grafana";
;
"2022-04-01 21:54:09,stdout,logger=settings t=2022-04-01T21:54:09.88+0000 lvl=info msg=""Config overridden from Environment variable"" var=""GF_EXPRESSIONS_ENABLED=false""";
;
"2022-04-01 21:54:09,stdout,logger=settings t=2022-04-01T21:54:09.88+0000 lvl=info msg=""Config overridden from Environment variable"" var=""GF_PLUGINS_PLUGIN_ADMIN_ENABLED=false""";
;
"2022-04-01 21:54:09,stdout,logger=settings t=2022-04-01T21:54:09.88+0000 lvl=info msg=""Config overridden from Environment variable"" var=""GF_PLUGINS_ALLOW_LOADING_UNSIGNED_PLUGINS=panodata-map-panel""";
;
"2022-04-01 21:54:09,stdout,logger=settings t=2022-04-01T21:54:09.88+0000 lvl=info msg=""Config overridden from Environment variable"" var=""GF_METRICS_ENABLED=false""";
;
"2022-04-01 21:54:09,stdout,logger=settings t=2022-04-01T21:54:09.88+0000 lvl=info msg=""Config overridden from Environment variable"" var=""GF_EXPLORE_ENABLED=false""";
;
"2022-04-01 21:54:09,stdout,logger=settings t=2022-04-01T21:54:09.88+0000 lvl=info msg=""Config overridden from Environment variable"" var=""GF_ALERTING_ENABLED=false""";
;
"2022-04-01 21:54:09,stdout,logger=settings t=2022-04-01T21:54:09.88+0000 lvl=info msg=""Config overridden from Environment variable"" var=""GF_USERS_ALLOW_SIGN_UP=false""";
;
"2022-04-01 21:54:09,stdout,logger=settings t=2022-04-01T21:54:09.88+0000 lvl=info msg=""Config overridden from Environment variable"" var=""GF_DASHBOARDS_DEFAULT_HOME_DASHBOARD_PATH=/var/lib/grafana-static/dashboards/vwsfriend/VWsFriend/overview.json""";
;
"2022-04-01 21:54:09,stdout,logger=settings t=2022-04-01T21:54:09.88+0000 lvl=info msg=""Config overridden from Environment variable"" var=""GF_SECURITY_ADMIN_PASSWORD=*********""";
;
"2022-04-01 21:54:09,stdout,logger=settings t=2022-04-01T21:54:09.88+0000 lvl=info msg=""Config overridden from Environment variable"" var=""GF_SECURITY_ADMIN_USER=*************""";
;
"2022-04-01 21:54:09,stdout,logger=settings t=2022-04-01T21:54:09.88+0000 lvl=info msg=""Config overridden from Environment variable"" var=""GF_SERVER_ENABLE_GZIP=true""";
;
"2022-04-01 21:54:09,stdout,logger=settings t=2022-04-01T21:54:09.88+0000 lvl=info msg=""Config overridden from Environment variable"" var=""GF_SERVER_HTTP_PORT=3000""";
;
"2022-04-01 21:54:09,stdout,logger=settings t=2022-04-01T21:54:09.88+0000 lvl=info msg=""Config overridden from Environment variable"" var=""GF_PATHS_PROVISIONING=/etc/grafana/provisioning""";
;
"2022-04-01 21:54:09,stdout,logger=settings t=2022-04-01T21:54:09.88+0000 lvl=info msg=""Config overridden from Environment variable"" var=""GF_PATHS_PLUGINS=/var/lib/grafana/plugins""";
;
"2022-04-01 21:54:09,stdout,logger=settings t=2022-04-01T21:54:09.88+0000 lvl=info msg=""Config overridden from Environment variable"" var=""GF_PATHS_LOGS=/var/log/grafana""";
;
"2022-04-01 21:54:09,stdout,logger=settings t=2022-04-01T21:54:09.88+0000 lvl=info msg=""Config overridden from Environment variable"" var=""GF_PATHS_DATA=/var/lib/grafana""";
;
"2022-04-01 21:54:09,stdout,logger=settings t=2022-04-01T21:54:09.88+0000 lvl=info msg=""Config overridden from command line"" arg=""default.log.mode=console""";
;
"2022-04-01 21:54:09,stdout,logger=settings t=2022-04-01T21:54:09.88+0000 lvl=info msg=""Config overridden from command line"" arg=""default.paths.provisioning=/etc/grafana/provisioning""";
;
"2022-04-01 21:54:09,stdout,logger=settings t=2022-04-01T21:54:09.88+0000 lvl=info msg=""Config overridden from command line"" arg=""default.paths.plugins=/var/lib/grafana/plugins""";
;
"2022-04-01 21:54:09,stdout,logger=settings t=2022-04-01T21:54:09.88+0000 lvl=info msg=""Config overridden from command line"" arg=""default.paths.logs=/var/log/grafana""";
;
"2022-04-01 21:54:09,stdout,logger=settings t=2022-04-01T21:54:09.88+0000 lvl=info msg=""Config overridden from command line"" arg=""default.paths.data=/var/lib/grafana""";
;
"2022-04-01 21:54:09,stdout,logger=settings t=2022-04-01T21:54:09.88+0000 lvl=info msg=""Config loaded from"" file=/etc/grafana/grafana.ini";
;
"2022-04-01 21:54:09,stdout,logger=settings t=2022-04-01T21:54:09.88+0000 lvl=info msg=""Config loaded from"" file=/usr/share/grafana/conf/defaults.ini";
;
"2022-04-01 21:54:09,stdout,logger=settings t=2022-04-01T21:54:09.88+0000 lvl=warn msg=""falling back to legacy setting of 'min_interval_seconds'";" please use the configuration option in the `unified_alerting` section if Grafana 8 alerts are enabled."""
;
"2022-04-01 21:54:09,stdout,logger=settings t=2022-04-01T21:54:09.86+0000 lvl=info msg=""Starting Grafana"" version=8.4.3 commit=b7d2911cac branch=HEAD compiled=2022-03-02T13:01:27Z";
;
2022-04-01 21:54:05,stdout,;
;
2022-04-01 21:54:05,stdout,Please restart Grafana after installing plugins. Refer to Grafana documentation for instructions if necessary.;
;
2022-04-01 21:54:05,stdout,;
;
2022-04-01 21:54:05,stdout,âœ” Downloaded panodata-map-panel v0.16.0 zip successfully;
;
2022-04-01 21:54:03,stdout,;
;
2022-04-01 21:54:03,stdout,Please restart Grafana after installing plugins. Refer to Grafana documentation for instructions if necessary.;
;
2022-04-01 21:54:03,stdout,;
;
2022-04-01 21:54:03,stdout,âœ” Downloaded gapit-htmlgraphics-panel v2.0.3 zip successfully;
;
2022-04-01 21:54:02,stdout,;
;
2022-04-01 21:54:02,stdout,Please restart Grafana after installing plugins. Refer to Grafana documentation for instructions if necessary.;
;
2022-04-01 21:54:02,stdout,;
;
2022-04-01 21:54:02,stdout,âœ” Downloaded natel-plotly-panel v0.0.7 zip successfully;
;
2022-04-01 21:54:00,stdout,;
;
2022-04-01 21:54:00,stdout,Please restart Grafana after installing plugins. Refer to Grafana documentation for instructions if necessary.;
;
2022-04-01 21:54:00,stdout,;
;
2022-04-01 21:54:00,stdout,âœ” Downloaded pr0ps-trackmap-panel v2.1.2 zip successfully;
;
2022-04-01 21:53:58,stdout,;
;
2022-04-01 21:53:58,stdout,Please restart Grafana after installing plugins. Refer to Grafana documentation for instructions if necessary.;
;
2022-04-01 21:53:58,stdout,;
;
2022-04-01 21:53:58,stdout,âœ” Downloaded marcusolsson-json-datasource v1.3.1 zip successfully;
`

**Reply by tillsteinbach** [reply]
I will try to make it a bit easier with release 0.16.6 that I try to release tonight. Then you only have to set the desired ports but nothing else to achieve what you want.

**Reply by sodgaard** [reply]
Thanks. What had I done wrong?
Should I do a complete reinstall, or can I install only parts to keep the data collected?

/Stephen

**Reply by sodgaard** [reply]
In the updates pushed 10 hrs ago I still see that eg. Grafana is only identified as Grafana instead of eg GrafanaVW. This calls for a conflict with eg. Teslamate.

**Reply by tillsteinbach** [reply]
is it really? Usually the container name is prefixed by the stock name.
The 0.16.6 mainly allows to set the ports in the env file. So no Manipulation in the compose file should be necessary.

**Reply by sodgaard** [reply]
> is it really? Usually the container name is prefixed by the stock name.
> 
It is not which is why it broke my working TeslaMate installation.

> The 0.16.6 mainly allows to set the ports in the env file. So no Manipulation in the compose file should be necessary.
But do you reckon that I need to reinstall the full VW package?
Did you identify why on my current install VW connect isworking, but grafana only shows a blank screen/time out...

/Stephen

**Reply by tillsteinbach** [reply]
Is the folder in which the teslamate docker-compose is located also named docker? If yes, can you rename one?

**Reply by tillsteinbach** [reply]
Alternativeliy you can also set the stacks name with something like `docker-compose --project-name vwsfriend up` or even in the env file like `COMPOSE_PROJECT_NAME=vwsfriend`

**Reply by sodgaard** [reply]
Here are the running containers side by side Teslamate & VWsFriend:

![image](https://user-images.githubusercontent.com/44952678/161704410-eeb118b6-bab8-4066-96f3-e8393d0cb0d0.jpeg)

**Reply by sodgaard** [reply]
In /volume1/@docker/volumes I have the following:
![image](https://user-images.githubusercontent.com/44952678/161705913-e6917ea9-c51a-4adf-9589-bf100d393b02.png)

**Reply by tillsteinbach** [reply]
Did you now use the new version and change the ports using the environment variables?
```
VWSFRIEND_PORT=4010
GF_SERVER_HTTP_PORT=3010
```

**Reply by sodgaard** [reply]
> Did you now use the new version and change the ports using the environment variables?
> 
> ```
> 
> VWSFRIEND_PORT=4010
> 
> GF_SERVER_HTTP_PORT=3010
> 
> ```

No, this is the 3 day old install. I am still awaiting your response if I need to do a full reinstall including erasing the DB?

**Reply by tillsteinbach** [reply]
No, you just need to update the docker-compose file and do a `docker-compose pull`.

**Reply by sodgaard** [reply]
So got it up and running now. Thanks.
One issue was that I did not know that the docker-compose command prefixed the service group with the current folder name. I simply had copied the two files to the docker root directory for execution meaning that both teslamate and VWsFriend were prefixed with docker and thus collided. It works with alternate mapped ports now as well, so I have them running simultaneously. :-)

**Comment by sodgaard** [comment, marked as answer]
See above
